Sources/Identity.swift

```diff
@@ -36,19 +36,22 @@ import Foundation
         registerListener(type: EventType.edgeIdentity, source: EventSource.removeIdentity, listener: handleRemoveIdentity)
         registerListener(type: EventType.genericIdentity, source: EventSource.requestReset, listener: handleRequestReset)
         registerListener(type: EventType.hub, source: EventSource.sharedState, listener: handleHubSharedState)
-
-        // attempt to bootup
-        if state.bootupIfReady() {
-            createXDMSharedState(data: state.identityProperties.toXdmData(), event: nil)
-        }
-
     }
 
     public func onUnregistered() {
     }
 
     public func readyForEvent(_ event: Event) -> Bool {
-        return true
+        if state.hasBooted {
+            return true
+        }
+
+        if state.bootupIfReady(event: event, getSharedState: getSharedState(extensionName:event:barrier:)) {
```

Comment:
It is required to pass in the `barrier` version of the getSharedState API? You should be able to just `getSharedState(extensionName:event:)` for the standard shared state.